paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -344,15 +344,33 @@ def create_instance_active_requests_scaling_rule(
             )[{moving_average_window}s:]
         )
     """
+
+    # The prometheus HPA adapter needs kube_deployment and kube_namespace labels attached to the metrics its scaling on.
+    # The envoy-based metrics have no labels corresponding to the k8s resources, so we can add them in.
     metrics_query = f"""
-        {desired_instances} / {current_replicas}
+        label_replace(
+            label_replace(
+                {desired_instances} / {current_replicas}, 
+                "kube_deployment", "{deployment_name}", "", ""
+            ),
+            "kube_namespace", "{namespace}", "", ""
+        )
+    """
+    seriesQuery = f"""
```

Comment:
nit: this should be `series_query`

Reply:
I am not a python developer clearly 🤦‍♂️